src/main/java/org/embl/mobie/io/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -343,17 +334,32 @@ private double[] getXYZScale( OmeZarrMultiscales.CoordinateTransformations coord
         double[] xyzScale = null;
         if (scale != null && zarrAxesList != null) {
             int scalesFirstIndexBackward = scale.length - 1;
-            if (zarrAxes.containsXYZCoordinates()) {
-                xyzScale = new double[]{
-                        scale[scalesFirstIndexBackward],
-                        scale[scalesFirstIndexBackward - 1],
-                        scale[scalesFirstIndexBackward - 2] };
+            if (zarrAxes.hasChannels()) {
+                if (zarrAxes.is3DWithChannels() || zarrAxes.is2D()) {
+                    xyzScale = new double[]{
+                            scale[scalesFirstIndexBackward],
+                            scale[scalesFirstIndexBackward - 1],
+                            1.0
+                    };
+                } else {
+                    xyzScale = new double[]{
+                            scale[scalesFirstIndexBackward],
+                            scale[scalesFirstIndexBackward - 1],
+                            scale[scalesFirstIndexBackward - 2]};
+                }
             } else {
-                xyzScale = new double[]{
-                        scale[scalesFirstIndexBackward],
-                        scale[scalesFirstIndexBackward - 1],
-                        1.0
-                };
+                if (zarrAxes.containsXYZCoordinates()) {
```

Comment:
what does `containsXYZCoordinates()`mean?
does that mean `is3D()`?

Reply:
@KateMoreva could you have a look here? I agree with @tischi that `containsXYZCoordinates` is confusing and `is3D` would be more intuitive.

Reply:
renamed